### PR TITLE
Convert the `as` and `is` operators into their own Expr subclasses.

### PR DESCRIPTION
**Pull request by harlanhaskins**
There's more duplication than I'd like, but it's pretty simple.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
Tests passed.

File: Sources/Utility Passes/JavaScriptGen.swift [hunk 1]
```diff
@@ -135,14 +135,19 @@ class JavaScriptGen<StreamType: TextOutputStream>: ASTTransformer {
     stream.write("return ")
     visit(expr.value)
   }
+
+  override func visitCoercionExpr(_ expr: CoercionExpr) {
+    visit(expr.rhs)
```
Comment (segiddins) [inline Sources/Utility Passes/JavaScriptGen.swift]: Pretty sure this needs to be the lhs?

Comment (harlanhaskins) [reply Sources/Utility Passes/JavaScriptGen.swift]: Yikes. Thanks!

**Comment by harlanhaskins** [comment]
@segiddins Good to merge?

File: Sources/Base/ASTContext.swift [hunk 2]
```diff
@@ -762,7 +762,10 @@ public class ASTContext {
     }
   }
   
-  func canCoerce(_ type: DataType, to other: DataType) -> Bool {
+  func canCoerce(_ _type: DataType, to _other: DataType) -> Bool {
```
Comment (segiddins) [inline Sources/Base/ASTContext.swift]: since the raw args are never used, why do they need to have the underscored names?

**Review by segiddins — APPROVED** [review]
Other than the comment about the args, looks good!

File: Sources/Base/ASTContext.swift (hunk 2 above)
Comment (harlanhaskins) [reply Sources/Base/ASTContext.swift]: Oh, y'know what, I forgot that I could just shadow their names...

**Comment by harlanhaskins** [comment]
⛵️

**Pull request merged**